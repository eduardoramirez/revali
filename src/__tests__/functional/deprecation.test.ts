import {getIntrospectionQuery, graphql, printSchema} from 'graphql'
import 'jest'

import {compileSchema, Field, ObjectType, TSGraphQLString} from 'revali/index'

describe('Deprecation', () => {
  it('Generates correct deprecation comments and directives', async () => {
    @ObjectType()
    class SampleObject {
      @Field()
      public normalField!: string

      @Field({deprecationReason: 'sample object field deprecation reason'})
      public describedField!: string

      @Field({
        type: TSGraphQLString,
        deprecationReason: 'sample object getter field deprecation reason',
      })
      get describedGetterField(): string {
        return 'describedGetterField'
      }

      @Field({
        type: TSGraphQLString,
        deprecationReason: 'sample object method field deprecation reason',
      })
      public methodField(): string {
        return 'methodField'
      }
    }

    @ObjectType()
    class Query {
      @Field({type: TSGraphQLString})
      public normalQuery(): string {
        return 'normalQuery'
      }

      @Field({type: TSGraphQLString, deprecationReason: 'sample query deprecation reason'})
      public describedQuery(): string {
        return 'describedQuery'
      }

      @Field({type: SampleObject})
      public object(): SampleObject {
        return new SampleObject()
      }
    }

    @ObjectType()
    class Mutation {
      @Field({type: TSGraphQLString})
      public normalMutation(): string {
        return 'normalMutation'
      }

      @Field({type: TSGraphQLString, deprecationReason: 'sample mutation deprecation reason'})
      public describedMutation(): string {
        return 'describedMutation'
      }
    }

    const schema = compileSchema({Query, Mutation})
    const {errors} = await graphql(schema, getIntrospectionQuery())
    expect(errors).toBeFalsy()
    expect(printSchema(schema)).toMatchSnapshot()
  })
})
