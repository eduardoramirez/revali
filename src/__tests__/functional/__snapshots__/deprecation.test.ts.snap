// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deprecation Generates correct deprecation comments and directives 1`] = `
"type Mutation {
  normalMutation: String!
  describedMutation: String! @deprecated(reason: \\"sample mutation deprecation reason\\")
}

type Query {
  normalQuery: String!
  describedQuery: String! @deprecated(reason: \\"sample query deprecation reason\\")
  object: SampleObject!
}

type SampleObject {
  normalField: String!
  describedField: String! @deprecated(reason: \\"sample object field deprecation reason\\")
  describedGetterField: String! @deprecated(reason: \\"sample object getter field deprecation reason\\")
  methodField: String! @deprecated(reason: \\"sample object method field deprecation reason\\")
}
"
`;
